src/analyzer/sonogram.cpp

```diff
@@ -0,0 +1,98 @@
+/* 
+   Strawberry Music Player
+   This file was part of Clementine.
+   Copyright 2004, Melchior FRANZ <mfranz@kde.org>
+   Copyright 2009-2010, David Sansome <davidsansome@gmail.com>
+   Copyright 2010, 2014, John Maguire <john.maguire@gmail.com>
+   Copyright 2014-2015, Mark Furneaux <mark@furneaux.ca>
+   Copyright 2014, Krzysztof Sobiecki <sobkas@gmail.com>
+
+   Strawberry is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   Strawberry is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with Strawberry.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/* Original Author:  Melchior FRANZ  <mfranz@kde.org>  2004
+ */
+
+#include "sonogram.h"
+
+#include <QPainter>
+
+using Analyzer::Scope;
+
+const char* Sonogram::kName =
+    QT_TRANSLATE_NOOP("AnalyzerContainer", "Sonogram");
+
+Sonogram::Sonogram(QWidget* parent)
+    : Analyzer::Base(parent, 9), scope_size_(128) {}
+
+Sonogram::~Sonogram() {}
+
+void Sonogram::resizeEvent(QResizeEvent* e) {
+  QWidget::resizeEvent(e);
+
+// only for gcc < 4.0
+#if !(__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 0))
+  resizeForBands(height() < 128 ? 128 : height());
+#endif
+
+  canvas_ = QPixmap(size());
+  canvas_.fill(palette().color(QPalette::Window));
+}
+
+void Sonogram::analyze(QPainter& p, const Scope& s, bool new_frame) {
```

Comment:
```suggestion
void Sonogram::analyze(QPainter &p, const Scope &s, bool new_frame) {
```
- Please adapt same code style for the remaining code.
- Newline between function and code here and everywhere else.